packages/hash/shared/src/types.ts

```diff
@@ -1,393 +1,14 @@
-
-export type PrimitiveDataTypeKey = keyof typeof primitiveDataTypes;
-
-export type PrimitiveDataTypeTitle =
-  typeof primitiveDataTypes[PrimitiveDataTypeKey]["title"];
-
-type TypeDefinition = {
-  title: string;
-  description?: string;
-};
-
-type EntityTypeDefinition = TypeDefinition & { entityTypeId: VersionedUri };
-
-type PropertyTypeDefinition = TypeDefinition & { propertyTypeId: VersionedUri };
-
-type DataTypeDefinition = TypeDefinition & { dataTypeId: VersionedUri };
-
-type LinkEntityTypeDefinition = TypeDefinition & {
-  linkEntityTypeId: VersionedUri;
-};
-
-type TypeDefinitions = {
-  entityType: Record<SystemEntityTypeKey, EntityTypeDefinition>;
-  linkEntityType: Record<SystemLinkEntityTypeKey, LinkEntityTypeDefinition>;
-  propertyType: Record<SystemPropertyTypeKey, PropertyTypeDefinition>;
-  dataType: Record<PrimitiveDataTypeKey, DataTypeDefinition>;
-};
-
-/**
- * The system and block protocol types that are statically available at run-time.
- */
-export const types: TypeDefinitions = {
-  entityType: Object.entries(systemEntityTypes).reduce(
-    (prev, [key, { title, description }]) => {
-      const definition: EntityTypeDefinition = {
-        title,
-        description,
-        entityTypeId: generateSystemTypeId({ kind: "entity-type", title }),
-      };
-
-      return { ...prev, [key]: definition };
-    },
-    {} as Record<SystemEntityTypeKey, EntityTypeDefinition>,
-  ),
-  propertyType: Object.entries(systemPropertyTypes).reduce(
-    (prev, [key, { title, description }]) => {
-      const definition: PropertyTypeDefinition = {
-        title,
-        description,
-        propertyTypeId: generateSystemTypeId({
-          kind: "property-type",
-          title,
-        }),
-      };
-
-      return { ...prev, [key]: definition };
-    },
-    {} as Record<SystemPropertyTypeKey, PropertyTypeDefinition>,
-  ),
-  dataType: Object.entries(primitiveDataTypes).reduce(
-    (prev, [key, { title, description }]) => {
-      const definition: DataTypeDefinition = {
-        title,
-        description,
-        dataTypeId: generateBlockProtocolTypeId({
-          kind: "data-type",
-          title,
-        }),
-      };
-
-      return { ...prev, [key]: definition };
-    },
-    {} as Record<PrimitiveDataTypeKey, DataTypeDefinition>,
-  ),
-  linkEntityType: Object.entries(systemLinkEntityTypes).reduce(
-    (prev, [key, { title, description }]) => {
-      const definition: LinkEntityTypeDefinition = {
-        title,
-        description,
-        linkEntityTypeId: generateSystemTypeId({
-          kind: "entity-type",
-          title,
-        }),
-      };
-
-      return { ...prev, [key]: definition };
-    },
-    {} as Record<SystemLinkEntityTypeKey, LinkEntityTypeDefinition>,
-  ),
-};
+// Since we already use a template string here, we will keep it. When we
+// brand these types, this may not be needed anymore.
+export type VersionedUri = `${string}/v/${number}`;
+export type BaseUri = string;
```

Comment:
Similar feeling as expressed above, I'm not sure we should be redefining these types when they're defined in an upstream package.

**_If_** we decide branding in a public library API is a bad idea, maybe we can look at re-exporting them here and shadowing with branded versions, but until then I'd leave them out.

Reply:
(even though `BaseUri` is one of the painful ones without branding)